compiler/rustc_codegen_ssa/src/mir/rvalue.rs

```diff
@@ -222,10 +224,20 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
                     bug!("Found {operand_kind:?} for operand {operand:?}");
                 };
                 if let OperandValueKind::Immediate(out_scalar) = cast_kind {
-                    let cast_bty = bx.backend_type(cast);
-                    Some(OperandValue::Immediate(Self::transmute_immediate(
-                        bx, imm, in_scalar, out_scalar, cast_bty,
-                    )))
+                    match (in_scalar, out_scalar) {
```

Comment:
I could certainly push the `OperandValueKind::Immediate` down to the two arms.

I'm not sure if I like it better, though, since it would makes the "these match" the arms are doing a bit less obvious.  How strongly do you feel about it?  Anything I missed that'd be a better layout?